### PR TITLE
[sc115424] add send messages by batch to sqs sdk

### PR DESCRIPTION
**Pull request by mtakerrabet**
add send messages by batch to to an aws sqs queue

### CODE REVIEW

File: subprojects/micronaut-amazon-awssdk-sqs/src/main/java/com/agorapulse/micronaut/amazon/awssdk/sqs/groovy/SimpleQueueServiceExtensions.java [hunk 1]
```diff
@@ -69,4 +72,39 @@ public static String sendMessage(
...
+    /**
+     * Sends messages with additional configuration into the default queue.
+     *
+     * @param messagesList          the list of the messages bodies with their  Ids
+     * @param messageConfiguration additional configuration
+     * @return message id
+     */
+    public static List<String> sendMessages(
+        SimpleQueueService self,
+        Map<String, String> messagesList,
```
Comment (musketyr) [inline subprojects/micronaut-amazon-awssdk-sqs/src/main/java/com/agorapulse/micronaut/amazon/awssdk/sqs/groovy/SimpleQueueServiceExtensions.java:84]: why is `messagesList` a map?

Comment (musketyr) [reply subprojects/micronaut-amazon-awssdk-sqs/src/main/java/com/agorapulse/micronaut/amazon/awssdk/sqs/groovy/SimpleQueueServiceExtensions.java:84]: it's worth renaming `messages` or `messagesByIds`

**Review by musketyr — CHANGES_REQUESTED** [review]
looks good, just a minor thing: I think using the `messagesList` for a `Map` is not a good practice. can you rename it.

I think it would make sense to to also add the support into `@QueueClient` declarative clients. e.g. by using `@Batch` on a `Map<String, Object>` parameters, what do you think?

another subject is if there is a need to backport the change also into AWS SDK v1 library. so far we were trying to keep these in sync.

**Comment by mtakerrabet** [comment]
> looks good, just a minor thing: I think using the `messagesList` for a `Map` is not a good practice. can you rename it.
> 
> I think it would make sense to to also add the support into `@QueueClient` declarative clients. e.g. by using `@Batch` on a `Map<String, Object>` parameters, what do you think?
> 
> another subject is if there is a need to backport the change also into AWS SDK v1 library. so far we were trying to keep these in sync.

Hi @musketyr, thanks for the review, I renamed the parameter and did the modification for the `@QueueClient` declarative clients client, but I didn't the changes on the  AWS SDK v1 library, the `SendMessageBatchRequest` is provided by the `software.amazon.awssdk:sqs` which is not embedded in

**Review by wmedjek — APPROVED** [review]

File: subprojects/micronaut-amazon-awssdk-sqs/src/main/java/com/agorapulse/micronaut/amazon/awssdk/sqs/groovy/SimpleQueueServiceExtensions.java (hunk 1 above)
Comment (musketyr) [reply subprojects/micronaut-amazon-awssdk-sqs/src/main/java/com/agorapulse/micronaut/amazon/awssdk/sqs/groovy/SimpleQueueServiceExtensions.java:84]: thanks, I will take a look and made some tests

**Comment by musketyr** [comment]
@mtakerrabet please, check this update to your PR: https://github.com/agorapulse/micronaut-aws-sdk/pull/199/commits/cd862c244f2a50c942182ceeeec39f64170132fd.

So the number of messages in the batch has its hard limit? In that case we can partition the messages directly in the introduced method.

**Comment by mtakerrabet** [comment]
> @mtakerrabet please, check this update to your PR: [cd862c2](https://github.com/agorapulse/micronaut-aws-sdk/commit/cd862c244f2a50c942182ceeeec39f64170132fd).
> 
> So the number of messages in the batch has its hard limit? In that case we can partition the messages directly in the introduced method.

Thank you for the updates @musketyr, it looks good to me, 
and for the number of messages in the batch, yes, sqs set a hard limit to 10 messages by request

**Pull request closed without merging**